storage_service/locations/models/package.py

```diff
@@ -1426,6 +1426,33 @@ def create_replicas(self):
         for replicator_loc in replicator_locs:
             self.replicate(replicator_loc)
 
+    def _replace_callback_placeholders(self, uri, body):
+        """
+        Replaces post store callback placeholders with values.
+
+        Currently, this method replaces the following placeholders:
+        - <package_uuid>: Replaced with package UUID
+        - <package_name>: Replaced with package name, with trailing UUID
+        removed. For AIPs created directly from a transfer, this will be
+        equivalent to the post-sanitization Transfer name.
+        """
+        current_package = os.path.split(self.current_path)[1]
+        current_package_root = os.path.splitext(current_package)[0]
+        # Split on extension twice if package compressed as .tar.gz
+        if current_package_root.endswith(".tar"):
+            package_name_with_uuid = os.path.splitext(current_package_root)[0]
+        else:
+            package_name_with_uuid = current_package_root
+        package_name = package_name_with_uuid[:-37]
+
+        uri = uri.replace("<package_uuid>", self.uuid).replace(
+            "<package_name>", package_name
+        )
+        body = body.replace("<package_uuid>", self.uuid).replace(
+            "<package_name>", package_name
+        )
```

Comment:
It's common in Python to find these expressive ways of doing things. I find it hard but it's fun. Perhaps, you could use a list comprehension, e.g.:

```python
return [
    item.replace("<package_uuid>", self.uuid).replace("<package_name>", self.name)
    for item in (uri, body)
]
```

Reply:
```python
repls = ("<package_uuid>", "UUID"), ("<package_name>", "NAME")
return [
    reduce(lambda a, kv: a.replace(*kv), repls, item)
    for item in (uri, body)
]
```

I actually find these solutions so hard to read! 🤣

Reply:
I find the lambda hard to read (I love an arrow function in JS, but in Python anonymous functions still seem weird to me 😂) so went with the list comprehension. Thanks!